### PR TITLE
Fix no cache user for user re-fetching

### PR DESCRIPTION
**Pull request by zaiste**

### CODE REVIEW

**Review by mmiszy — CHANGES_REQUESTED** [review]
Could you also update `@saleor/auth-sdk` to the latest version?

File: apps/storefront/lib/useUser.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { useUserQuery } from "@/saleor/api";
 
 export const useUser = () => {
-  const { data, loading } = useUserQuery();
+  const { data, loading } = useUserQuery({ fetchPolicy: "no-cache" });
```
Comment (mmiszy) [inline apps/storefront/lib/useUser.ts:4]: What about other queries that may depend on the user being logged in, for example `checkout`?

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: apps/storefront/lib/useUser.ts (hunk 1 above)
Comment (zaiste) [reply apps/storefront/lib/useUser.ts]: @mmiszy changed to `cache-and-network`

File: Dockerfile.base [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM node:16-slim
 WORKDIR /app
 
 # Setup pnpm package manager
-RUN npm install -g pnpm@7.11.0
+RUN npm install -g pnpm
```
Comment (mmiszy) [inline Dockerfile.base:5]: You should pin the version to avoid accidentally installing pnpm 9 and breaking the project

**Review by mmiszy — CHANGES_REQUESTED** [review]

**Review by mmiszy — APPROVED** [review]

**Pull request merged**